wordpress-mobile/WordPress-Android
# Issue/10726 site pages author picker phase iv

Fixes part of #10726 

**Description**
This is the fourth PR in a series of five that introduce author filtering on the pages view. 

This PR handles showing the author display name on the pages list, including while viewing search results, when the author filter is set to everyone. When the filter is set to ME, the author name will not show.

Me - no author name | Everyone - author name
--------|------- 
<kbd><img src="https://user-images.githubusercontent.com/506707/84429929-f6d9e100-abf6-11ea-88ff-ad309586372e.png" width=320></kbd> | <kbd><img src="https://user-images.githubusercontent.com/506707/84429934-f9d4d180-abf6-11ea-9a4e-0e7fde5cafd9.png" width=320></kbd>

**Details**
- Added `author` property to Published, Drafts, Scheduled, and Trashed data models in `PageItem`
- Added a new function in `PageItemViewHolder` to manage the formatting of the subtitle line to include the author
- Added a new placeholder string for formatting a subtitle with three parameters
- Updated `PageListViewModel` to pass author name when filter is not ME
- Added two tests for showing-not showing author name

**Notes**

- Repeating my original call for opinions on the following behavior: The Posts feature also shows an author filter and saves state in App prefs. I implemented Pages to use the same appPrefs property, so the filter selection is maintained across pages to posts. 
- There is a “todo” placeholder in `PagesViewModel` . Setting up tracking will occur in the next PR.
    
**To test:**
Prep

- Be an author on a wp.com site that you are not the owner on
- Have pages in the above site in each of the four statuses: Published, Scheduled, Drafts, and Trashed (1) that you authored (2) that others authored

(A)

- Launch WPAndroid
- Select the site you prepped for the test
- Navigate to Pages
- Select Me from the author filter
- Ensure the author name is not shown
- Select Everyone from the author filter
- Ensure the author name is shown

(B)

- Launch WPAndroid
- Select the site you prepped for the test
- Navigate to Pages
- Select Me from the author filter
- Search for a page that you authored
- Ensure the author name is not shown
- Select Everyone from the author filter
- Search for any page
- Ensure the author name is shown


PR submission checklist:
